api.go

```diff
@@ -36,6 +36,9 @@ type FileMatch struct {
 	// Ranking; the higher, the better.
 	Score float64 // TODO - hide this field?
 
+	// Experimental. A score vector. Components should be compared pairwise.
+	ranks []float64
```

Comment:
should we document if vectors have different lengths? eg if we update scoring functions, some shards will likely have different vector sizes in the ranks.